engine/time/src/main/java/io/deephaven/time/calendar/CalendarDay.java

```diff
@@ -117,7 +120,11 @@ public boolean isInclusiveEnd() {
      * @return length of the day in nanoseconds
      */
     public long businessNanos() {
-        return businessTimeRanges.stream().map(TimeRange::nanos).reduce(0L, Long::sum);
+        if (businessNanos < 0) {
+            businessNanos = Arrays.stream(businessTimeRanges).map(TimeRange::nanos).reduce(0L, Long::sum);
+        }
+
+        return businessNanos;
```

Comment:
We should ideally only be reading a volatile variable once; can store in local variable, and when computing also store in a local variable.